linetools/analysis/utils.py

```diff
@@ -43,3 +43,47 @@ def box_ew(spec):
 
     # Return
     return EW, sigEW
+
+#Gaussian EW
+def gaussian_ew(spec):
+    """  EW calculation using Gaussian fit
+    Observer frame, not rest-frame
+    wvlim must be set!
+    spec must be set!
+
+    Parameters
+    ----------
+    spec -- Tuple of (wave, fx, sig)
+
+    Returns:
+    ----------
+    EW, sigEW : EW and error in observer frame
+    Note: Tested in test_absline_anly
+    """
+    # Grab
+    wv,fx,sig = spec
+
+    # dwv
+    dwv = wv - np.roll(wv,1)
+    dwv[0] = dwv[1]
+    
+    # Fit the data using a Gaussian
+    # Initial guesses
+    amp_init = np.mean(fx)/2. #half the mean flux
+    stddev_init = 3*np.mean(dwv) #3 pixels
+    mean_init = np.mean(wv) #half wave range
+    # Model initialization
+    g_init = models.GaussianAbsorption1D(amplitude=amp_init.value, mean=mean_init.value, stddev=stddev_init.value)
+    # Fitting initialization
+    fit_g = fitting.LevMarLSQFitter()
+    # Actual fit
+    g = fit_g(g_init, wv, fx, weights=1./sig)
+
+    # Area under curve of Gaussian is [amplitude*stdev*sqrt(2*pi)]
+    EW = g.amplitude.value * g.stddev.value * np.sqrt(2 * np.pi) #unitless
+    EW = EW * wv.unit #add the same unit as wv
```

Comment:
It seems that these implemented Gaussian models from astropy.modeling do not support units...